include/simdjson/simdutf8check_westmere.h

```diff
@@ -181,9 +176,9 @@ really_inline void check_utf8<Architecture::WESTMERE>(
   if ((_mm_testz_si128(_mm_or_si128(in.v0, in.v1), high_bit)) == 1) {
     // it is ascii, we just check continuation
     state.has_error =
-        _mm_or_si128(_mm_cmpgt_epi8(state.previous.carried_continuations,
+        _mm_or_si128(_mm_cmpgt_epi8(state.previous.first_len,
                                     _mm_setr_epi8(9, 9, 9, 9, 9, 9, 9, 9, 9, 9,
-                                                  9, 9, 9, 9, 9, 1)),
+                                                  9, 9, 9, 3, 2, 1)),
```

Comment:
I did some fuzzy tests, found no error, but I'm still not convinced by that line. I found no difference with keeping "9, 9", and I think this is suspicious. I'll have to do more tests before the merge.

Reply:
If only we had a fuzzer tool for such tasks.

Reply:
:-)

Reply:
In my understanding of the range algorithm, "9,9,9,9,9,9,9,9,9,9,9,9,9,9,9,1" should not detect any error if the third or the second byte before the end of previous is in the range 0xF0~0xFF and the current 16 bytes are full ascii. Right now I'm trying to be sure we test that case.

I wonder what are the probabilities our fuzzer can generate that case.

Reply:
So the line was in fact wrong, but the fuzzer had no difficulty to generate the problematic documents. And I'm a bit ashamed to write down what I was doing wrong.